arangod/Cluster/DBServerAgencySync.cpp

```diff
@@ -157,14 +174,42 @@ DBServerAgencySyncResult DBServerAgencySync::execute() {
   }
 
   auto& clusterInfo = _server.getFeature<ClusterFeature>().clusterInfo();
-  uint64_t planIndex = 0;
-  auto plan = clusterInfo.getPlan(planIndex);
+  uint64_t planIndex = 0, currentIndex = 0;
+
+
+  // The rationale is that even with 1000 databases and us waking up every 5 seconds,
+  // we should approximately visit every database without being dirty once per hour.
+  // That is why we divide by 720.
+  auto moreDirt = mfeature.pickRandomDirty(
+    static_cast<size_t>(std::ceil(static_cast<double>(mfeature.lastNumberOfDatabases()) / 720.)));
+  auto dirty = mfeature.dirty();
+  // Add `moreDirt` to `dirty` but remove from `moreDirt`, if already dirty anyway.
+  // Then we can reasonably be surprised if we find anything in a database in `moreDirt`.
+  for (auto it = moreDirt.begin(); it != moreDirt.end();) {
+    if (dirty.find(*it) != dirty.end()) {
+      dirty.insert(*it);
+      it++;
+    } else {
+      it = moreDirt.erase(it);
+    }
+  }
+
 
-  if (plan == nullptr) {
+  if (dirty.empty()) {
+    LOG_TOPIC("0a62f", DEBUG, Logger::MAINTENANCE)
+      << "DBServerAgencySync::execute no dirty collections";
+    result.success = true;
```

Comment:
We are returning an error message but also set `success` to `true` here.
This is a bit counter-intuitive, but the same happens a few lines below. So I am wondering if this is intentional or not.
If it is intentional, we should leave a code comment why encountering an error allows us to still set `success` to true here.

Reply:
again, not an error but a message